src/data/matrix/notation.lean

```diff
@@ -130,6 +130,130 @@ cons_val_succ x u 0
   vec_cons x u i = x :=
 by { fin_cases i, refl }
 
+/-! ### Numeral (`bit0` and `bit1`) indices
+The following definitions and `simp` lemmas are to allow any
+numeral-indexed element of a vector given with matrix notation to
+be extracted by `simp` (even when the numeral is larger than the
+number of elements in the vector, which is taken modulo that number
+of elements by virtue of the semantics of `bit0` and `bit1` and of
+addition on `fin n`).
+-/
+
+/-- `vec_join ho u v` joins two vectors of lengths `m` and `n` to produce
+one of length `o = m + n`.  `ho` provides control of definitional equality
+for the vector length. -/
+def vec_join (ho : o = m + n) (u : fin m → α) (v : fin n → α) : fin o → α :=
```

Comment:
Should this go to `data/fin`?

Reply:
I don't know. And if it should go there, what name would fit best with `data/fin` naming conventions?

Reply:
By analogy to `fin.cons` and `list.append`, how about `fin.append`?

Reply:
I've now renamed and moved this.